recirq/qml_lfe/circuit_blocks.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2020 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Sycamore native circuit building blocks."""
+
+import os
+import cirq
+import cirq_google
+import functools
+
+
+@functools.lru_cache(maxsize=128)
+def _load_circuit(fname):
+    with open(os.path.join(os.path.dirname(__file__), fname), "r") as f:
+        return cirq.read_json(f)
+
+
+def tsym_block(qubits, params):
+    """Create a tsym block.
+
+    Y axis rotations followed by a real entried permutation operation.
+
+    Args:
+        qubits: The pair of qubits to apply this operation on.
+        params: The length 2 list of Y rotation angles.
+
+    Returns:
+        A list of `cirq.Operations` realizing a tsym block.
+    """
+    mapped_circuit = _load_circuit("tsym_permute.txt").transform_qubits(
+        {cirq.GridQubit(0, 0): qubits[0], cirq.GridQubit(0, 1): qubits[1]}
+    )
+    rots = [cirq.Y(qubits[0]) ** params[0], cirq.Y(qubits[1]) ** params[1]]
+    return rots + list(mapped_circuit.all_operations())
+
+
+def _get_op(code):
```

Comment:
What is going on here?  Maybe this is worth a comment?

Reply:
This is used to generate the single qubit gates for the scrambling blocks just like we did in: https://www.nature.com/articles/s41586-019-1666-5.pdf